controller/controller.py

```diff
@@ -523,32 +523,30 @@ def runStagedNmap(self, iprange, discovery=True, stage=1, stop=False):
 		if not stop:
 			textbox = self.view.createNewTabForHost(str(iprange), 'nmap (stage '+str(stage)+')', True)
 			outputfile = self.logic.runningfolder+"/nmap/"+getTimestamp()+'-nmapstage'+str(stage)		
-			
-			if stage == 1:												# webservers/proxies
-				ports = self.settings.tools_nmap_stage1_ports
-			elif stage == 2:											# juicy stuff that we could enumerate + db
-				ports = self.settings.tools_nmap_stage2_ports
-			elif stage == 3:											# bruteforceable protocols + portmapper + nfs
-				ports = self.settings.tools_nmap_stage3_ports
-			elif stage == 4:											# first 30000 ports except ones above
-				ports = self.settings.tools_nmap_stage4_ports
-			else:														# last 35535 ports
-				ports = self.settings.tools_nmap_stage5_ports
 
 			command = "nmap "
 			if not discovery:											# is it with/without host discovery?
 				command += "-Pn "
-			command += "-T4 -sV "										# without scripts (faster)
-			if not stage == 1:
-				command += "-n "										# only do DNS resolution on first stage
-			if os.geteuid() == 0:										# if we are root we can run SYN + UDP scans
-				command += "-sSU "
-				if stage == 2:
-					command += "-O "									# only check for OS once to save time and only if we are root otherwise it fails
+			if "stage"+str(stage) in self.settings.stagedNmapSettings.keys():
+				if "flags" in self.settings.stagedNmapSettings["stage"+str(stage)].keys():
+					command += self.settings.stagedNmapSettings['stage'+str(stage)]['flags'] + ' '
+				else:
+
+					command += self.settings.stagedNmapSettings['stage0']['flags'] + ' '
+				if "ports" in self.settings.stagedNmapSettings["stage"+str(stage)].keys():
+					if self.settings.stagedNmapSettings['stage'+str(stage)]['ports'] == "-": # Check for all ports
+						command += "-p"+self.settings.stagedNmapSettings['stage'+str(stage)]['ports']
+					else:
+						command += "-p "+self.settings.stagedNmapSettings['stage'+str(stage)]['ports']
+				else:
+					command += "-p "+self.settings.stagedNmapSettings['stage0']['ports']
 			else:
-				command += "-sT "
-			command += "-p "+ports+' '+iprange+" -oA "+outputfile
-							
+				command += self.settings.stagedNmapSettings['stage'+str(stage)]['flags'] + ' ' + "-p " + \
+                           self.settings.stagedNmapSettings['stage0']['ports']
```

Comment:
Is this part of the else block? I'm not much of a python programmer but the tabbing here is a little unclear to me and just wanted to confirm it was right

Reply:
Disregard, I just realized its part of word wrap :P